jax/_src/scipy/special.py

```diff
@@ -1010,3 +1011,78 @@ def lpmn_values(m: int, n: int, z: jnp.ndarray, is_normalized: bool) -> jnp.ndar
   l_max = n
 
   return _gen_associated_legendre(l_max, z, is_normalized)
+
+
+
+@partial(jit, static_argnums=(4,))
+def _sph_harm(m: jnp.ndarray,
+              n: jnp.ndarray,
+              theta: jnp.ndarray,
+              phi: jnp.ndarray,
+              n_max: int) -> jnp.ndarray:
+  """Computes the spherical harmonics."""
+
+  len_inputs = m.shape[0]
+  harmonics = jnp.zeros((len_inputs), dtype=jnp.complex64)
```

Comment:
Just realized that these two lines are no longer necessary, as `harmonics` is defined from scratch below.

Also, I think given the new implementation, vectorized versions of inputs should work correctly. You could test this by passing things like
```
n = jnp.arange(3, 6)
m = jnp.arange(-3, 3)[:, None]
theta = 0.0
phi = jnp.pi
out = sph_harm(n, m, theta, phi)
```
This should output a (6, 3) array

Reply:
Done. Thanks!